modules/_labs/pagination/react/lib/GoTo.tsx

```diff
@@ -0,0 +1,70 @@
+import styled from '@emotion/styled';
+import {type} from '@workday/canvas-kit-react-core';
+import TextInput from '@workday/canvas-kit-react-text-input';
+import React from 'react';
+
+const GoToLabel = styled('div')({
+  margin: 'auto',
+  paddingRight: '8px',
+  ...type.body,
+  ...type.variant.label,
+});
+
+const GoToWrapper = styled('div')({
+  display: 'flex',
+  paddingLeft: '1px',
+});
+
+const InputWrapper = styled('div')({
+  minWidth: '10px',
+  maxWidth: '45px',
+});
+
+const GoTo: React.FC<{
+  onSubmit: (page: number, e: React.SyntheticEvent) => void;
+  max: number;
+}> = props => {
+  const {onSubmit, max} = props;
+  const [value, setValue] = React.useState('');
+
+  const validatePage = (text: string) => {
+    const textAsInteger = parseInt(text, 10);
+    if (textAsInteger < 1) {
+      return 0;
+    }
+    if (textAsInteger > max) {
+      return 0;
+    }
+    return textAsInteger;
+  };
+  return (
+    <GoToWrapper>
+      <GoToLabel>Go To</GoToLabel>
+      <InputWrapper>
+        <TextInput
+          data-testid="goToPage"
+          width={53}
+          height={32}
+          min={1}
+          max={max}
+          value={value}
+          type="number"
+          aria-label="Go to"
```

Comment:
I think the `GoToLabel` should be a `label` element with a `for`/`htmlFor` pointing to this `input`'s `id`. An ID can be computed or passed. We have examples where it can be passed in or defaults to a `uuid`. This has the added benefit of clicking the label text focuses the input. This isn't required for the PR to be merged, but needs to be done before this component moves out of labs. It is also a good practice. Connecting the input to the label will remove the need for the `aria-label` attribute.